In [2]:
import pandas as pd

### data load

In [3]:
# original data
data_path = '/Users/okchang/mainbiz/project/p1_final/data/original_data.csv'
df = pd.read_csv(data_path)
df.info()

/var/folders/hy/0q0xn9_j1m3ddxnt492l8q240000gn/T/ipykernel_35584/3306291301.py:3: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(data_path)


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 475236 entries, 0 to 475235
Data columns (total 39 columns):
 #   Column     Non-Null Count   Dtype  
---  ------     --------------   -----  
 0   상가업소번호     475236 non-null  object 
 1   상호명        475236 non-null  object 
 2   지점명        9520 non-null    object 
 3   상권업종대분류코드  475236 non-null  object 
 4   상권업종대분류명   475236 non-null  object 
 5   상권업종중분류코드  475236 non-null  object 
 6   상권업종중분류명   475236 non-null  object 
 7   상권업종소분류코드  475236 non-null  object 
 8   상권업종소분류명   475236 non-null  object 
 9   표준산업분류코드   475145 non-null  object 
 10  표준산업분류명    475145 non-null  object 
 11  시도코드       475236 non-null  int64  
 12  시도명        475236 non-null  object 
 13  시군구코드      475236 non-null  int64  
 14  시군구명       475236 non-null  object 
 15  행정동코드      475236 non-null  int64  
 16  행정동명       475236 non-null  object 
 17  법정동코드      475236 non-null  int64  
 18  법정동명       475236 non-null  object 
 19  지번코드       475236 non-n

In [6]:
# columns 확인
print(len(df.columns))
df.columns

39


Index(['상가업소번호', '상호명', '지점명', '상권업종대분류코드', '상권업종대분류명', '상권업종중분류코드',
       '상권업종중분류명', '상권업종소분류코드', '상권업종소분류명', '표준산업분류코드', '표준산업분류명', '시도코드',
       '시도명', '시군구코드', '시군구명', '행정동코드', '행정동명', '법정동코드', '법정동명', '지번코드',
       '대지구분코드', '대지구분명', '지번본번지', '지번부번지', '지번주소', '도로명코드', '도로명', '건물본번지',
       '건물부번지', '건물관리번호', '건물명', '도로명주소', '구우편번호', '신우편번호', '동정보', '층정보',
       '호정보', '경도', '위도'],
      dtype='object')

In [8]:
# data 확인
df.iloc[500]

상가업소번호            MA010120220805446886
상호명                               맥스수학
지점명                                NaN
상권업종대분류코드                           P1
상권업종대분류명                            교육
상권업종중분류코드                         P105
상권업종중분류명                         일반 교육
상권업종소분류코드                       P10501
상권업종소분류명                       입시·교과학원
표준산업분류코드                        P85501
표준산업분류명                        일반 교과학원
시도코드                                11
시도명                              서울특별시
시군구코드                            11740
시군구명                               강동구
행정동코드                         11740540
행정동명                              명일2동
법정동코드                       1174010100
법정동명                               명일동
지번코드               1174010100100480012
대지구분코드                               1
대지구분명                               대지
지번본번지                               48
지번부번지                             12.0
지번주소               서울특별시 강동구 명일동 48-12
도로명코드                    

## 데이터 정리

1. 불필요한 컬럼 제거: (475236, 39) -> (475145, 10)
    - 상호명
    - 지점명
    - 상권업종소분류명: 업종 선택을 위함 -> 표쥰산업분류명로만 부족 -> 카페
    - 표쥰산업분류명: 업종 선택을 위함 -> 커피 전문점 (+ 결측치 있는 행 제거)
    - 행정동 
    - 법정동
    - 지번주소
    - 도로명주소
    - 경도
    - 위도 
2. 업종 선택(카페): (475145, 10) -> (21425, 10)
    - 상권업종소분류명 -> 카페
    - 표쥰산업분류명 -> 커피 전문점

### 1) 불필요한 컬럼 제거

In [9]:
columns_to_keep = ['상호명', '지점명', '상권업종소분류명', '표준산업분류명', '행정동명', '법정동명', '지번주소', '도로명주소', '경도', '위도']

df_filtered = df[columns_to_keep]
print(df_filtered.columns)
df_filtered.info()

Index(['상호명', '지점명', '상권업종소분류명', '표준산업분류명', '행정동명', '법정동명', '지번주소', '도로명주소',
       '경도', '위도'],
      dtype='object')
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 475236 entries, 0 to 475235
Data columns (total 10 columns):
 #   Column    Non-Null Count   Dtype  
---  ------    --------------   -----  
 0   상호명       475236 non-null  object 
 1   지점명       9520 non-null    object 
 2   상권업종소분류명  475236 non-null  object 
 3   표준산업분류명   475145 non-null  object 
 4   행정동명      475236 non-null  object 
 5   법정동명      475236 non-null  object 
 6   지번주소      475236 non-null  object 
 7   도로명주소     475236 non-null  object 
 8   경도        475236 non-null  float64
 9   위도        475236 non-null  float64
dtypes: float64(2), object(8)
memory usage: 36.3+ MB


In [10]:
print(df.shape, df_filtered.shape)

(475236, 39) (475236, 10)


### 2) 업종 필터링
- 표준산업분류명 == 커피 전문점 or 상권업종소분류명 == 카페

In [14]:
cafe_df = df_filtered[(df_filtered['표준산업분류명'] == '커피 전문점') | (df_filtered['상권업종소분류명'] == '카페')]
cafe_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 21436 entries, 10 to 475225
Data columns (total 10 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   상호명       21436 non-null  object 
 1   지점명       1252 non-null   object 
 2   상권업종소분류명  21436 non-null  object 
 3   표준산업분류명   21425 non-null  object 
 4   행정동명      21436 non-null  object 
 5   법정동명      21436 non-null  object 
 6   지번주소      21436 non-null  object 
 7   도로명주소     21436 non-null  object 
 8   경도        21436 non-null  float64
 9   위도        21436 non-null  float64
dtypes: float64(2), object(8)
memory usage: 1.8+ MB


### 3) 결측치 제거
- 업종정보에 결측치있는 데이터 제거

In [24]:
# 결측치 확인
missing_cols = cafe_df.columns[cafe_df.isnull().any()].tolist()
print(missing_cols)
print(cafe_df[cafe_df['표준산업분류명'].isnull()].shape)
print(cafe_df[cafe_df['표준산업분류명'].isnull()]['상호명']) 
# 보면 카페 아님 -> 제거

['지점명', '표준산업분류명']
(11, 10)
201228        도아코퍼레이션
409990          다방구밴드
420233       한국커피브레이크
420924      할리스커피주협의회
422972       대한수제커피협회
426528       투썸플레이스역삼
439160      아시아국제커피협회
442708         통합커피학회
461190    매머드커피랩점주연합회
463588     컴포즈커피제1호조합
464706       한국커피검정협회
Name: 상호명, dtype: object


In [25]:
cafe_df.dropna(subset=['표준산업분류명'], inplace=True)
print(cafe_df[cafe_df['표준산업분류명'].isnull()].shape) # 확인: 제거됨

(0, 10)


/var/folders/hy/0q0xn9_j1m3ddxnt492l8q240000gn/T/ipykernel_35584/3169492377.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  cafe_df.dropna(subset=['표준산업분류명'], inplace=True)


In [27]:
cafe_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 21425 entries, 10 to 475225
Data columns (total 10 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   상호명       21425 non-null  object 
 1   지점명       1250 non-null   object 
 2   상권업종소분류명  21425 non-null  object 
 3   표준산업분류명   21425 non-null  object 
 4   행정동명      21425 non-null  object 
 5   법정동명      21425 non-null  object 
 6   지번주소      21425 non-null  object 
 7   도로명주소     21425 non-null  object 
 8   경도        21425 non-null  float64
 9   위도        21425 non-null  float64
dtypes: float64(2), object(8)
memory usage: 1.8+ MB


In [28]:
# 정리된 데이터 저장
save_path = '/Users/okchang/mainbiz/project/p1_final/data'
filename = 'cafe_seoul.csv'
cafe_df.to_csv(f'{save_path}/{filename}', index=False, encoding='utf-8-sig')

In [29]:
# 확인
saved_csv = pd.read_csv(f'{save_path}/{filename}')
saved_csv.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21425 entries, 0 to 21424
Data columns (total 10 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   상호명       21425 non-null  object 
 1   지점명       1250 non-null   object 
 2   상권업종소분류명  21425 non-null  object 
 3   표준산업분류명   21425 non-null  object 
 4   행정동명      21425 non-null  object 
 5   법정동명      21425 non-null  object 
 6   지번주소      21425 non-null  object 
 7   도로명주소     21425 non-null  object 
 8   경도        21425 non-null  float64
 9   위도        21425 non-null  float64
dtypes: float64(2), object(8)
memory usage: 1.6+ MB


## 정리

- 데이터 크기: 259.2MB(475236, 39) -> 4MB(21425, 10)

- 서울 외 다른 지역 정보는 사이트에서 다운받아오면 됨

- 카페가 아닌 다른 업종을 고르려면 이 노트북에서 업종컬럼 필터링

